RummikubGameApp/RummikubGameApp/GameLogic.cs

```diff
@@ -0,0 +1,340 @@
+        }
+
+        public Tile[] StartGame()
+        {
+            // Initialize the game by creating the tiles and shuffling them
+            
+            InitializeTiles();
+            Shuffle();
+            Tile[] arr = Deal();
+            // Set the first player as the current player
+            currentPlayerIndex = 0;
+            return arr;
+        }
+
+
+        public void InitializeTiles()
+        {
+            int index = 0;
+            for (int i = 1; i <= 13; i++)
+            {
+                tiles[index++] = new Tile { Color = "Red", Value = i };
+                tiles[index++] = new Tile { Color = "Red", Value = i };
+                tiles[index++] = new Tile { Color = "Blue", Value = i };
+                tiles[index++] = new Tile { Color = "Blue", Value = i };
+                tiles[index++] = new Tile { Color = "Yellow", Value = i };
+                tiles[index++] = new Tile { Color = "Yellow", Value = i };
+                tiles[index++] = new Tile { Color = "Black", Value = i };
+                tiles[index++] = new Tile { Color = "Black", Value = i };
+            }
+            tiles[index++] = new Tile { Color = "Joker", Value = 0 };
+            tiles[index++] = new Tile { Color = "Joker", Value = 14 };
+        }
+        public Tile[] Shuffle()
+        {
+            // Shuffle the tiles randomly
+            Random random = new Random();
+            for (int i = 0; i < tiles.Length; i++)
+            {
+                int j = random.Next(i, tiles.Length);
+                Tile temp = tiles[i];
+                tiles[i] = tiles[j];
+                tiles[j] = temp;
+            }
+            return tiles;
+        }
+
+       
+        public Tile[] Deal()
+        {
+            // Deal 14 tiles from the deck to start the game
+            for (int i = 0; i < 14; i++)
+            {
+                hand[i] = tiles[i];
+            }
+            return hand;
+        }
+        public Tile[] DealAI()
+        {
+            // Deal 14 tiles from the deck to start the game
+            for (int i = 0; i < 14; i++)
+            {
+                AIhand[i] = tiles[i+14];
+            }
+            return hand;
+        }
+
+
+        public Tile[] bynumber()
+        {
+            int count = 0;
+            while (hand[count] != null)
+                count++;
+            for (int i = 0; i < count - 1; i++)
+                for (int j = 0; j < count - i - 1; j++)
+                    if (hand[j].Value > hand[j + 1].Value)
+                    {
+                        // swap temp and arr[i]
+                        Tile temp = hand[j];
+                        hand[j] = hand[j + 1];
+                        hand[j + 1] = temp;
+                    }
+            return hand;
+        }
+
+        public Tile[] bycolor()
+        {
+            hand = bynumber();
+            int n = 0;
+            while (hand[n] != null)
+                n++;
+            int count = 0;

```

Comment:
the name count is not a good name since it isn't used to count, but to find an index.